commit-patch-buffer.el

```diff
@@ -84,16 +90,30 @@ DIRECTORY with commit-patch(1)."
      `((log-edit-listfun . (lambda () ',patch-files)))
      "*commit*")))
 
-(defun commit-patch-buffer (&optional arg)
+(defun commit-patch-buffer (&optional arg amend)
   "Commit the patch in the current buffer, applying it to the
 repository in the appropriate directory with commit-patch(1). If
 the current buffer is not in diff-mode or ARG is non-nil then it
 will ask interactively which buffer to commit and to which
-directory to commit it."
+directory to commit it.  If AMEND is non-nil, we amend the
+previous commit instead of creating a new one."
   (interactive "P")
   (if (and (not arg) (eq major-mode 'diff-mode))
-      (commit-patch-buffer-in-directory (buffer-name) (autodetect-patch-directory-root))
-    (call-interactively 'commit-patch-buffer-in-directory)))
+      (commit-patch-buffer-in-directory (buffer-name) (autodetect-patch-directory-root) amend)
+
+    (let ((current-prefix-arg amend))
+      (call-interactively 'commit-patch-buffer-in-directory))))
+
+(defun commit-patch-buffer-amend (&optional arg)
+  "Commit the patch in the current buffer, applying it to the
+repository in the appropriate directory with commit-patch(1). If
+the current buffer is not in diff-mode or ARG is non-nil then it
+will ask interactively which buffer to commit and to which
+directory to commit it.  This is identical to
+`commit-patch-buffer' except it amends the last commit by default
+instead of creating a new one."
+  (interactive "P")
```

Comment:
Yes. It allows a prefix argument to affect commit-patch-buffer-amend in the same way it affects commit-patch-buffer.